### ISSUE TITLE
Search Results (non-fuzzy) Sort by Match Making

### ISSUE THREAD

**Issue by ktomk** [zealdocs/zeal#859]
In the non-fuzzy search result (the non-experimental one), it would be great to show the results ordered by match within title based on the number of characters before (preferable low) and behind (preferable low, but not as much as before).

That wouldn't change anything from the type of search just in which order the results are listed, showing direct matches first.

Weighting the before and behind non-matching characters should provide a good result. E.g. a direct match (no characters behind and before) would be preferred (when searching multiple collections even if unstable this should present results pretty well, esp. with IDE integrations).

Some others might chime in for a simple weight distribution for more than zero characters before and after (e.g. "_" might have a special weight than other characters), however, displaying the more direct matches first looks like an easy win to me.

So far I need to inspect the result list on the highlighted parts to get this job done, but why not a (simple?) sort by Zeal instead?

I'm using an Intellij Idea plugin to open up Zeal, an example search query it creates is: "`php,wordpress,drupal,zend,laravel,yii,joomla,ee,codeigniter,cakephp,phpunit,symfony,typo3,twig,smarty,phpp:substr`".  Results (non-fuzzy) currently look like this:

![image](https://user-images.githubusercontent.com/352517/34501380-f0eed184-f00e-11e7-9265-83c437395f7a.png)

The entries on top are just not what has been looking for sort of obviously, the results are displayed alphabetically. Maybe sort UI elements can be added to support this as this feature request changes learned user experience.

Fuzzy search btw. gets it pretty well, but also displays results far off.

In summary suggestion so far: Sort of results non-alphabetically unless results do have the same weight:

* Full match is highest
* Match at the start is pretty valuable
* Match at the end is too, most often a little less than at the beginning

Commonly languages have separator sequences like `_` or `::` or `.` or `\` or `/` that could constitute a full word match (most languages are ASCII based which's character classes are pretty well known).

**Comment by trollixx** [comment]
This is fixed in v0.5.0.

**Comment by ktomk** [comment]
Oh, that was quick. Let me try to get that version...

**Comment by trollixx** [comment]
Duplicate of #679.

**Comment by ktomk** [comment]
Thanks a lot for your support! I have to admit I was not able to find the duplicate, next time I need to search harder.

**Comment by trollixx** [comment]
No worries :)

**Comment by ktomk** [comment]
Why should I worry. Never seen a Github issue responded to that fast. Was just fixing it's wording and then  already your answer. What could I worry about? :D Trying git clone and whatever now...

**Comment by trollixx** [comment]
I just happen to be here digging through issues. Usually my response time is waaaaay longer :P

**Comment by ktomk** [comment]
Can confirm fixed in v0.5.0:

![image](https://user-images.githubusercontent.com/352517/34503777-b531a458-f01a-11e7-96c8-d6a0238a99f2.png)
